--- hpklinux-3.05.06/linux/lib/Makefile.am.orig	2006-10-18 05:55:44.000000000 +0200
+++ hpklinux-3.05.06/linux/lib/Makefile.am	2006-12-29 22:43:57.820938845 +0100
@@ -21,7 +21,7 @@
 ##
 
 HPI_DIR = $(top_srcdir)/hpi
-AM_CFLAGS = -O3 -pipe -Wall -DHPI_DEBUG @HPI_64BIT@
+AM_CFLAGS = -pipe -Wall -DHPI_DEBUG @HPI_64BIT@
 ##INCLUDES = -I/usr/src/linux/include -I$(HPI_DIR)
 INCLUDES = -I$(HPI_DIR)
 
@@ -33,6 +33,7 @@
                    $(HPI_DIR)/hpifunc.c\
                    $(HPI_DIR)/hpidebug.c
 libhpi_la_LDFLAGS = -version-info $(INTERFACE_CURRENT):$(INTERFACE_REVISION):$(INTERFACE_AGE)
+libhpi_la_LIBADD = -lm
 
 CLEANFILES = *~ *.d
 MAINTAINERCLEANFILES = *~ Makefile.in
